contracts/interchain-token/HyperliquidInterchainToken.sol

```diff
@@ -9,13 +9,16 @@ import { IHyperliquidDeployer } from '../interfaces/IHyperliquidDeployer.sol';
 /**
  * @title HyperliquidInterchainToken
  * @notice This contract implements an interchain token with Hyperliquid-specific modifications.
- * @dev Inherits from HyperLiquidDeployer first to ensure slot 0 is properly reserved,
+ * @dev Inherits from HyperLiquidDeployer to use keccak256-based storage slots for deployer addresses,
  * then from InterchainToken for standard functionality.
  * This maintains the standard InterchainToken while providing Hyperliquid compatibility.
  */
 contract HyperliquidInterchainToken is HyperliquidDeployer, InterchainToken, IHyperliquidDeployer {
     error NotService(address caller);
 
+    /// bytes32(uint256(keccak256('hyperliquid-interchain-token-deployer')) - 1)
```

Comment:
This is wrong. The slot mentioned in hyperliquid [docs](https://hyperliquid.gitbook.io/hyperliquid-docs/for-developers/hyperevm/hypercore-less-than-greater-than-hyperevm-transfers#linking-core-and-evm-spot-assets) is `keccak256("HyperCore deployer")`. It seems like you're trying to follow the pattern from other contracts, but in this case, hyperliquid requires a specific slot calculation